templates/config/ekscluster-kfd-v1alpha2.yaml.tpl

```
@@ -98,156 +145,260 @@ spec:
       policy:
+        # This optional key is used to override automatic parameters
         overrides:
+          # This key is used to override the spec.distribution.common.nodeSelector setting
           nodeSelector: null
+          # This key is used to override the spec.distribution.common.tolerations setting
           tolerations: null
+          # This key is used to override some parameters on the ingresses managed by this module
           ingresses:
             gpm:
+              # if the authentication is enabled, it can be disabled
               disableAuth: false
+              # the host can be ovverridden, by default is gpm.{.spec.distribution.modules.ingress.baseDomain}
               host: ""
+              # the ingressClass can be overriden if needed
               ingressClass: ""
+        # configurations for the gatekeeper package
         gatekeeper:
+          # This parameter adds namespaces to the gatekeeper whitelist. These namespaces will
           additionalExcludedNamespaces: []
+      # This section contains all the configurations for the dr module
       dr:
+        # Configurations for the velero package
         velero:
+          # Velero configurations for EKS cluster
           eks:
+            # The S3 bucket that will be created
             bucketName: example-velero
-            iamRoleArn: arn:aws:iam::123456789012:role/example-velero
+            # This field is ignored, but needed. TBD better validation
+            iamRoleArn: dummyvalue
+            # The region where the bucket will be created (can be different from the overall region defined in .spec.region)
             region: eu-west-1
+        # This optional key is used to override automatic parameters
         overrides:
+          # This key is used to override the spec.distribution.common.nodeSelector setting
           nodeSelector: null
+          # This key is used to override the spec.distribution.common.tolerations setting
           tolerations: null
+      # This section contains all the configurations for the auth module
       auth:
+        # This optional key is used to override automatic parameters
         overrides:
+          # This key is used to override the spec.distribution.common.nodeSelector setting
           nodeSelector: null
+          # This key is used to override the spec.distribution.common.tolerations setting
+          tolerations: null
           ingresses:
             pomerium:
+              # the host can be ovverridden, by default is pomerium.{.spec.distribution.modules.ingress.baseDomain}
               host: ""
+              # the ingressClass can be overriden if needed
               ingressClass: ""
             dex:
+              # the host can be ovverridden, by default is login.{.spec.distribution.modules.ingress.baseDomain}
               host: ""
+              # the ingressClass can be overriden if needed
```

maybe we can say that the default is:
```suggestion
              # the ingressClass can be overridden if needed. Defaults to `internal` for dual-nginx and to `nginx` for single-nginx.
```

---

Let's skip the default for now